service/src/main/java/bio/terra/tanagra/service/authentication/JwtAccessTokenUtils.java

```diff
@@ -0,0 +1,95 @@
+package bio.terra.tanagra.service.authentication;
+
+import bio.terra.tanagra.utils.FileUtils;
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.exceptions.JWTVerificationException;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.auth0.jwt.interfaces.JWTVerifier;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.EncodedKeySpec;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+import org.bouncycastle.util.io.pem.PemObject;
+import org.bouncycastle.util.io.pem.PemReader;
+import org.springframework.util.Assert;
+
+public class JwtAccessTokenUtils {
+  private final JWTVerifier jwtVerifier;
+  private final ObjectMapper mapper = new ObjectMapper();
+  private final Base64.Decoder decoder = Base64.getUrlDecoder();
+
+  private static byte[] parsePEMFile(String pemFileName) throws IOException {
```

Comment:
nit: Consider moving the private methods to the bottom of the file, or after the public methods they support? I think it's a little easier to read the class when the constructor & public methods are at the top, since that's how someone would get into this class to begin with.

Reply:
done